src/main/kotlin/com/sourcegraph/cody/listeners/CodyDocumentListener.kt

```diff
@@ -30,37 +30,46 @@ class CodyDocumentListener(val project: Project) : BulkAwareDocumentListener {
   }
 
   override fun documentChangedNonBulk(event: DocumentEvent) {
-    // Can be called on non-EDT during IDE shutdown.
-    runInEdt {
-      val editor = FileEditorManager.getInstance(project).selectedTextEditor
-      if (editor?.document != event.document) {
-        return@runInEdt
-      }
+    try {
+      // Can be called on non-EDT during IDE shutdown.
+      runInEdt { handleDocumentEvent(event) }
+    } catch (e: IllegalStateException) {
+      // The error is thrown when a user opens the light bulb icon (from JetBrains).
+      // This event is not relevant to Cody (is not a change to the document), so we ignore it.
```

Comment:
I do not understand why invoking later should be banned on some threads 🫤 There are multiple errors like this reported in jetbrains youtrack (https://youtrack.jetbrains.com/issue/IDEA-300435/com.intellij.model.SideEffectGuardSideEffectNotAllowedException-Side-effect-not-allowed-INVOKELATER). Likely, it's a bug in jetbrains.